test/WETH9.symbolic.t.sol

```diff
@@ -0,0 +1,292 @@
+
+        uint256 galBalanceBefore = weth.balanceOf(gal);
+
+        vm.prank(guy);
+        weth.transferFrom(src, dst, wad);
+
+        uint256 galBalanceAfter = weth.balanceOf(gal);
+
+        assert(galBalanceAfter == galBalanceBefore);
+    }
+
+    // @dev transferFrom(src, dst, wad):
+    //      - decreases msg.sender's allowance by exactly wad.
+    function check_transferFrom_allowanceUpdate(address guy, address src, address dst, uint256 wad) public {
+        vm.assume(guy != src);
+        vm.assume(src != dst);
+        vm.deal(src, wad);
+        vm.prank(src);
+        weth.deposit{value: wad}();
+
+        uint256 guyAllowanceBefore = weth.allowance(src, guy);
+        vm.assume(guyAllowanceBefore != type(uint256).max);
+
+        vm.prank(guy);
+        weth.transferFrom(src, dst, wad);
+
+        uint256 guyAllowanceAfter = weth.allowance(src, guy);
+
+        assert(guyAllowanceAfter == guyAllowanceBefore - wad);
+    }
+
+    // @dev transferFrom(src, dst, wad):
+    //      - does not change allowance if caller is src.
+    function check_transferFrom_allowanceUpdate_callerIsSrc(address guy, address src, address dst, uint256 wad)
+        public
+    {
+        vm.assume(guy != src);
+        vm.assume(src != dst);
+        vm.deal(guy, wad);
+        vm.prank(guy);
+        weth.deposit{value: wad}();
+
+        uint256 guyAllowanceBefore = weth.allowance(guy, guy);
+        vm.assume(guyAllowanceBefore != type(uint256).max);
+
+        vm.prank(guy);
+        weth.transferFrom(guy, dst, wad);
+
+        uint256 guyAllowanceAfter = weth.allowance(guy, guy);
+
+        assert(guyAllowanceAfter == guyAllowanceBefore);
+    }
+
+    // @dev transferFrom(src, dst, wad):
+    //      - does not change msg.sender's allowance if set to type(uint256).max
+    function check_transferFrom_allowanceUpdate_maxAllowance(address guy, address src, address dst, uint256 wad)
+        public
+    {
+        vm.assume(src != dst);
+        vm.deal(src, wad);
+        vm.startPrank(src);
+        weth.deposit{value: wad}();
+        weth.approve(guy, type(uint256).max);
+        vm.stopPrank();
+
+        uint256 guyAllowanceBefore = weth.allowance(src, guy);
+
+        vm.prank(guy);
+        weth.transferFrom(src, dst, wad);
+
+        uint256 guyAllowanceAfter = weth.allowance(src, guy);
+
+        assert(guyAllowanceAfter == guyAllowanceBefore);
+        assert(guyAllowanceAfter == type(uint256).max);
+    }
+
+    function gen_calldata(bytes4 selector) internal returns (bytes memory) {
+        // Ignore view functions
+        // Skip for now
+
+        // Create symbolic values to be included in calldata
+        address guy = svm.createAddress("guy");
+        address src = svm.createAddress("src");
+        address dst = svm.createAddress("dst");
+        uint256 wad = svm.createUint256("wad");
+
+        // Generate calldata based on the function selector
+        bytes memory args;
+        if (selector == weth.withdraw.selector) {
```

Comment:
There aren't necessary since none of the WETH functions take dynamic-length args.